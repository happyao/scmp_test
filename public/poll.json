{
  "polls": [
    {
      "id": 1,
      "title": "Is bitcoin worth the time and money that mining requires?",
      "publishedDate": 1516605447,
      "answer": {
        "type": "Single",
        "options": [
          {
            "id": 1,
            "label": "Yes",
            "vote": 2
          },
          {
            "id": 2,
            "label": "No",
            "vote": 2
          }
        ]
      }
    },
    {
      "id": 2,
      "title": "Should chatbots replace humans in customer service jobs?",
      "publishedDate": 1516000647,
      "answer": {
        "type": "Single",
        "options": [
          {
            "id": 3,
            "label": "Yes",
            "vote": 4
          },
          {
            "id": 4,
            "label": "No",
            "vote": 2
          }
        ]
      }
    },
    {
      "id": 3,
      "title": "How are we feeling about 2018?",
      "publishedDate": 1515568647,
      "answer": {
        "type": "Single",
        "options": [
          {
            "id": 5,
            "label": "Hopeful",
            "vote": 2
          },
          {
            "id": 6,
            "label": "Doubtful",
            "vote": 2
          }
        ]
      }
    },
    {
      "id": 4,
      "title": "Which country/region have you ever visited? (Select all that applies)",
      "publishedDate": 1515482247,
      "answer": {
        "type": "Multi",
        "options": [
          {
            "id": 7,
            "label": "Hong Kong",
            "vote": 2
          },
          {
            "id": 8,
            "label": "China",
            "vote": 1
          },
          {
            "id": 9,
            "label": "Australia",
            "vote": 1
          },
          {
            "id": 10,
            "label": "Thailand",
            "vote": 1
          },
          {
            "id": 11,
            "label": "Korea",
            "vote": 1
          },
          {
            "id": 12,
            "label": "Japan",
            "vote": 3
          }
        ]
      }
    },
    {
      "id": 5,
      "title": "Will new benefits encourage you to study or work in mainland?",
      "publishedDate": 1515309447,
      "answer": {
        "type": "Single",
        "options": [
          {
            "id": 13,
            "label": "Yes",
            "vote": 2
          },
          {
            "id": 14,
            "label": "No",
            "vote": 1
          }
        ]
      }
    },
    {
      "id": 6,
      "title": "hi, hi another question 2?",
      "publishedDate": 1515309447,
      "answer": {
        "type": "Single",
        "options": [
          {
            "id": 14,
            "label": "Yes",
            "vote": 2
          },
          {
            "id": 15,
            "label": "No",
            "vote": 1
          }
        ]
      }
    },
    {
      "id": 7,
      "title": "hi another question?",
      "publishedDate": 1515309447,
      "answer": {
        "type": "Single",
        "options": [
          {
            "id": 16,
            "label": "Yes",
            "vote": 2
          },
          {
            "id": 17,
            "label": "No",
            "vote": 1
          }
        ]
      }
    },
    {
      "id": 8,
      "title": "hi another question 3?",
      "publishedDate": 1515309447,
      "answer": {
        "type": "Single",
        "options": [
          {
            "id": 18,
            "label": "Yes",
            "vote": 2
          },
          {
            "id": 19,
            "label": "No",
            "vote": 1
          }
        ]
      }
    },
    {
      "id": 9,
      "title": "hi another question 4?",
      "publishedDate": 1515309447,
      "answer": {
        "type": "Single",
        "options": [
          {
            "id": 20,
            "label": "Yes",
            "vote": 2
          },
          {
            "id": 21,
            "label": "No",
            "vote": 1
          }
        ]
      }
    }
  ]
}
