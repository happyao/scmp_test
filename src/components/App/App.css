.App {
  width: 100vw;
  background-color: #f6f6f6;
}
