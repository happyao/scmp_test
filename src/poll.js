var customData = {
  polls: [
    {
      id: 1,
      title: 'Is bitcoin worth the time and money that mining requires?',
      publishedDate: 1516605447,
      answer: {
        type: 'Single',
        options: [
          {
            id: 1,
            label: 'Yes',
            vote: 10
          },
          {
            id: 2,
            label: 'No',
            vote: 20
          }
        ]
      }
    },
    {
      id: 2,
      title: 'Should chatbots replace humans in customer service jobs?',
      publishedDate: 1516000647,
      answer: {
        type: 'Single',
        options: [
          {
            id: 3,
            label: 'Yes',
            vote: 30
          },
          {
            id: 4,
            label: 'No',
            vote: 40
          }
        ]
      }
    },
    {
      id: 3,
      title: 'How are we feeling about 2018?',
      publishedDate: 1515568647,
      answer: {
        type: 'Single',
        options: [
          {
            id: 5,
            label: 'Hopeful',
            vote: 50
          },
          {
            id: 6,
            label: 'Doubtful',
            vote: 60
          }
        ]
      }
    },
    {
      id: 4,
      title:
        'Which country/region have you ever visited? (Select all that applies)',
      publishedDate: 1515482247,
      answer: {
        type: 'Multi',
        options: [
          {
            id: 7,
            label: 'Hong Kong',
            vote: 70
          },
          {
            id: 8,
            label: 'China',
            vote: 80
          },
          {
            id: 9,
            label: 'Australia',
            vote: 90
          },
          {
            id: 10,
            label: 'Thailand',
            vote: 100
          },
          {
            id: 11,
            label: 'Korea',
            vote: 110
          },
          {
            id: 12,
            label: 'Japan',
            vote: 120
          }
        ]
      }
    },
    {
      id: 5,
      title: 'Will new benefits encourage you to study or work in mainland?',
      publishedDate: 1515309447,
      answer: {
        type: 'Single',
        options: [
          {
            id: 13,
            label: 'Yes',
            vote: 120
          },
          {
            id: 14,
            label: 'No',
            vote: 100
          }
        ]
      }
    }
  ]
}

module.exports = customData
