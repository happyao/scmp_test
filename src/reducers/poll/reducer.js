import { types } from '../../actions/types'
// import customData from '../../poll'

export const initialState = {
  currentID: 0,
  customData: []
}
export default (state = initialState, action) => {
  switch (action.type) {
    case types.GET_DATA:
      return { ...state, customData: action.customData }
    case types.CHOOSE_ITEM:
      return { ...state, currentID: action.id }
    case types.VOTE_ITEM:
      return {
        ...state,
        customData: state.customData.map(cd => {
          if (state.currentID === cd.id - 1) {
            return {
              ...cd,
              answer: {
                ...cd.answer,
                options: cd.answer.options.map(item => {
                  if (item.id === action.id) {
                    return { ...item, vote: item.vote + 1 }
                  } else {
                    return item
                  }
                })
              }
            }
          } else {
            return cd
          }
        })
      }

    default:
      return state
  }
}
