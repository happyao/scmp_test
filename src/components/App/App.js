import React, { Component } from 'react'
import { connect } from 'react-redux'
import './App.css'
import CurrentPoll from '../../containers/CurrentPoll'
import PollList from '../../containers/PollList'
import { getData } from '../../actions'
class App extends Component {
  componentDidMount () {
    // let xhr = new XMLHttpRequest()

    // xhr.open('GET', 'poll.json')
    // xhr.responseType = 'json'
    // xhr.send()
    // xhr.onerror = function () {
    //   alert(`Network Error`)
    // }

    // xhr.onload = function () {
    //   console.log(this)
    //   if (xhr.status !== 200) {
    //     alert(`Error ${xhr.status}: ${xhr.statusText}`)
    //   } else {
    //     let responseObj = xhr.response
    //     console.log(responseObj.polls)
    //     this.props.getData(responseObj.polls)
    //   }
    // }.bind(this)

    // xhr.timeout = 10000

    // we can also use ajax fetch
    this.getJson()
  }

  async getJson () {
    let response = await fetch('poll.json')
    let data = await response.json()
    this.props.getData(data.polls)
    console.log('getjson', data.polls)
  }

  render () {
    return (
      <div className='App'>
        {this.props.customData.length !== 0 && (
          <React.Fragment>
            <CurrentPoll />
            <PollList />
          </React.Fragment>
        )}
      </div>
    )
  }
}

const mapStateToProps = state => ({
  customData: state.poll.customData
})

const mapDispatchToProps = dispatch => ({
  getData: data => dispatch(getData(data))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)
export { App }
