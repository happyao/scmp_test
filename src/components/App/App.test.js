import React from 'react'
import { App } from './App'
import poll from '../../poll'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('App', () => {
  it('renders without crashing given the required props', () => {
    const props = {
      customData: poll.polls
    }

    const wrapper = shallow(<App {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
