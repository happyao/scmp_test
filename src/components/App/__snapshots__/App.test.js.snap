// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<div
  className="App"
>
  <Connect(CurrentPoll) />
  <Connect(PollList) />
</div>
`;
